src/libcore/mem.rs

```diff
@@ -65,6 +76,23 @@ pub use intrinsics::transmute;
 /// // Leak an I/O object, never closing the file
 /// let file = File::open("foo.txt").unwrap();
 /// mem::forget(file);
+///
+/// // The swap function uses forget
+/// fn swap<T>(x: &mut T, y: &mut T) {
+///     unsafe {
+///         // Give ourselves some scratch space to work with
+///         let mut t: T = uninitialized();
+///
+///         // Perform the swap, `&mut` pointers never alias
+///         ptr::copy_nonoverlapping(&*x, &mut t, 1);
+///         ptr::copy_nonoverlapping(&*y, x, 1);
+///         ptr::copy_nonoverlapping(&t, y, 1);
+///
+///         // y and t now point to the same thing, but we need to completely forget `t`
+///         // because it's no longer relevant.
```

Comment:
"because it's no longer relevant" is not a strong enough statement.  Maybe something like "we need to forget `t` because we do not want to run the destructor for `T` on its value, which is still owned somewhere outside this function." ?